modules/features/profile/src/test/java/au/com/shiftyjelly/pocketcasts/profile/ReferralsViewModelTest.kt

```diff
@@ -0,0 +1,150 @@
+package au.com.shiftyjelly.pocketcasts.profile
+
+import app.cash.turbine.test
+import au.com.shiftyjelly.pocketcasts.models.to.SignInState
+import au.com.shiftyjelly.pocketcasts.models.to.SubscriptionStatus
+import au.com.shiftyjelly.pocketcasts.models.type.SubscriptionFrequency
+import au.com.shiftyjelly.pocketcasts.models.type.SubscriptionPlatform
+import au.com.shiftyjelly.pocketcasts.models.type.SubscriptionTier
+import au.com.shiftyjelly.pocketcasts.models.type.SubscriptionType
+import au.com.shiftyjelly.pocketcasts.repositories.user.UserManager
+import au.com.shiftyjelly.pocketcasts.sharedtest.InMemoryFeatureFlagRule
+import au.com.shiftyjelly.pocketcasts.sharedtest.MainCoroutineRule
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.Feature
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.FeatureFlag
+import io.reactivex.Flowable
+import java.util.Date
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.whenever
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class ReferralsViewModelTest {
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
+
+    @get:Rule
+    val featureFlagRule = InMemoryFeatureFlagRule()
+
+    private val userManager: UserManager = mock()
+    private lateinit var viewModel: ReferralsViewModel
+    private val email = "support@pocketcasts.com"
+    private val statusAndroidPaidSubscription = SubscriptionStatus.Paid(
+        expiry = Date(),
+        autoRenew = true,
+        giftDays = 0,
+        frequency = SubscriptionFrequency.MONTHLY,
+        platform = SubscriptionPlatform.ANDROID,
+        subscriptionList = emptyList(),
+        type = SubscriptionType.PLUS,
+        tier = SubscriptionTier.PLUS,
+        index = 0,
+    )
+
+    @Before
+    fun setUp() {
+        FeatureFlag.setEnabled(Feature.REFERRALS, true)
+    }
```

Comment:
Would you mind adding a test also for a case when the FF is disabled for a paid account?

Reply:
Added in [356aa5f](https://github.com/Automattic/pocket-casts-android/pull/2746/commits/356aa5f28504032aa2ca405d955a8529face0c07)